lesson34/src/js/article.js

```diff
@@ -0,0 +1,159 @@
+  const article = createElementWithClassName("article", "article");
+  const articleTitle = createElementWithClassName("h1", "article__title");
+  const articleContent = createElementWithClassName("div", "article__content");
+  articleTitle.textContent = data.title;
+  articleContent.textContent = data.content;
+  articleFragment.appendChild(articleTitle).after(createArticleMeta(data));
+  data.thumbnail && articleFragment.appendChild(createArticleThumbnail(data));
+  articleFragment.appendChild(articleContent);
+  articleWrapper.appendChild(article).appendChild(articleFragment);
+}
+
+const addEventListenerForFavoriteButton = () => {
+  const favoriteButton = document.getElementById("js-favorite-button");
+  favoriteButton.addEventListener("click", (event) => {
+    event.preventDefault();
+    registerFavoriteButton();
+    setFavoriteArticleDataInLocalStorage();
+  });
+}
+
+const renderFavoriteButton = () => {
+  const favoriteButton = createElementWithClassName("button", "article__favorite-icon");
+  const img = document.createElement("img");
+  img.src = "./img/star-icon.svg";
+  favoriteButton.dataset.favoriteAction = "unregistered";
+  favoriteButton.id = "js-favorite-button";
+  articleWrapper.appendChild(favoriteButton).appendChild(img);
+}
+
+const createArticleMeta = ({ date }) => {
+  const articleMetaWrapper = createElementWithClassName("div", "article__meta");
+  const articleDate = createElementWithClassName("p", "article__date");
+  articleDate.textContent = date;
+  articleMetaWrapper.appendChild(articleDate);
+  return articleMetaWrapper;
+}
+
+const createArticleThumbnail = ({ thumbnail }) => {
+  const articleThumbnail = createElementWithClassName("p", "article__thumbnail");
+  const img = document.createElement("img");
+  img.src = thumbnail;
+  articleThumbnail.append(img);
+  return articleThumbnail;
+}
+
+const findArticleData = (data) => {
+  const urlParams = new URLSearchParams(window.location.search);
+  let result;
+  data.forEach(({ article }) => {
+    article.forEach((data) => data.id === urlParams.get("id") && (result = data));
+  });
+  return result;
+}
+
+const isRegisteredFavoriteArticle = () => {
+  const loginUserToken = localStorage.getItem("token");
+  const userFavoriteArticlesData = JSON.parse(localStorage.getItem("favoriteArticles"))?.[loginUserToken];
+  return userFavoriteArticlesData && userFavoriteArticlesData.some((data) => data[currentArticleData.id])
+
+}
+const init = async () => {
+  const api = "https://mocki.io/v1/caa7ab3a-f228-49f2-b522-4545cde742bd";
+  const responseData = await getArticleData(api);
+  const data = responseData?.data;
+  if (data) {
+    currentArticleData = findArticleData(data);
+    if (!currentArticleData) {
+      articleWrapper.textContent = "指定されたURLは存在しませんでした。";
+      return;
+    }
+    renderArticle(currentArticleData);
+    renderFavoriteButton();
+    addEventListenerForFavoriteButton();
+    isRegisteredFavoriteArticle() && registerFavoriteButton();
```

Comment:
` isRegisteredFavoriteArticle() && registerFavoriteButton();`
What are the special reasons for executing with &&?

Reply:
I wrote it this way because it is shorter and more concise than writing if statements.

Reply:
I see. It's wonderful!!Thank you very much！